src/engines/ngspice.c

```diff
@@ -94,17 +94,10 @@ static void ngspice_class_init (OreganoNgSpiceClass *klass)
 	object_class->finalize = ngspice_finalize;
 }
 
-static void ngspice_finalize (GObject *object)
-{
-	OreganoNgSpice *ngspice;
-	GList *iter;
-	int i;
-
-	ngspice = OREGANO_NGSPICE (object);
-	iter = ngspice->priv->analysis;
-	for (; iter; iter = iter->next) {
+void ngspice_analysis_finalize(GList *analysis) {
+	for (GList *iter = analysis; iter; iter = iter->next) {
```

Comment:
How did you compile this? I think you need C99+ to be able to do inline for loop declarations ey?